drools-game-engine-horserace-kjar/src/test/java/org/drools/game/horserace/tests/GameAPITest.java

```diff
@@ -0,0 +1,137 @@
+        Checkpoint checkpointone = new Checkpoint("CheckPointOne", startfinish);
+        initFacts.add( checkpointone );
+        Checkpoint checkpointtwo = new Checkpoint("CheckPointTwo", checkpointone);
+        initFacts.add( checkpointtwo );
+        Checkpoint checkpointthree = new Checkpoint("CheckPointThree", checkpointtwo);
+        initFacts.add( checkpointthree );
+        Checkpoint checkpointfour = new Checkpoint("CheckPointFour", checkpointthree);
+        initFacts.add( checkpointfour );
+        startfinish.setRequired(checkpointfour);
+        // Bootstrap the Game with the constructed house for this player
+        GameConfiguration gameConfiguration = new BaseGameConfigurationImpl( initFacts, "" );
+
+        game.bootstrap( gameConfiguration );
+
+        PlayerConfiguration playerConfiguration = new BasePlayerConfigurationImpl( null );
+
+        game.join( player, playerConfiguration );
+
+        game.execute( new EnterCheckpointCommand( player, startfinish ) );
+        game.execute( new EnterCheckpointCommand( player, checkpointone ) );
+        game.execute( new EnterCheckpointCommand( player, checkpointtwo ) );
+        game.execute( new EnterCheckpointCommand( player, checkpointthree ) );
+        game.execute( new EnterCheckpointCommand( player, checkpointfour ) );
+        game.execute( new EnterCheckpointCommand( player, startfinish ) );
+
+        List<GameMessage> messages = game.getAllMessages( player.getName() );
+        messages.addAll( game.getAllMessages( "system" ) );
+        
+        assertEquals( 15, messages.size() );
+
+        Set<String> messageTexts = messages.stream().map( m -> m.getText() ).collect( Collectors.toSet() );
+        
+        for(String string : messageTexts)
+        {
+            System.out.println(string);
+        }
+
+        //TODO: I could not get this working. I verified that the output is all correct manually. No idea what is going on.
+        assertThat( messageTexts,
+                Matchers.containsInAnyOrder(
+                        "Contestant Morton Abenthy Halputz has crossed a checkpoint!",
+                        "Contestant Morton Abenthy Halputz has crossed the finish line! Way to go! Now resetting the race for another attempt. ",
+                        "Contestant Morton Abenthy Halputz, mount your horse, get ready to race! Three... Two... One... aaaand... GO!",
+                        "You entered the checkpoint: StartFinish",
+                        "You entered the checkpoint: CheckPointOne",
+                        "You entered the checkpoint: CheckPointTwo",
+                        "You entered the checkpoint: CheckPointThree",
+                        "You entered the checkpoint: CheckPointFour",
+                        "Welcome, Morton Abenthy Halputz. Venture forth to the stables and mount a mighty steed!"
+                        ) );
+
+        Queue<Command> callbacks = game.getCallbacks();
+        //assertEquals( 8, callbacks.size() );
```

Comment:
add some mock callbacks here. Look at your Minecraft implementation to see what needs to be sent from the rules to Minecraft so the UI can be updated about the race status. I would say that probably, when you increment the score of each player, (the last checkpoint) you can send a callback to notify Minecraft that Player X just hit Checkpoint Y.